litgpt/finetune/lora.py

```diff
@@ -192,6 +193,8 @@ def main(fabric: L.Fabric, devices: int, seed: int, config: Config, data: LitDat
         copy_config_files(checkpoint_dir, save_path.parent)
         save_hyperparameters(setup, save_path.parent)
         save_prompt_style(data.prompt_style, save_path.parent)
+    fabric.barrier()
+    merge_lora(checkpoint_dir=save_path.parent)
```

Comment:
Why not do this on rank 0 only? merge_lora is single process. It would also limit the RAM use

Reply:
Yes, that's absolutely right.